packages/firestore/e2e/Transaction.e2e.js

```diff
@@ -335,5 +335,45 @@ describe('firestore.Transaction', () => {
       const snapshot = await docRef.get();
       snapshot.data().should.eql(jet.contextify(expected));
     });
+
+    it('should roll back any updates that failed', async () => {
+      const docRef = firebase.firestore().doc('v6/transactions/transaction/rollback');
+
+      await docRef.set({
+        turn: 0,
+      });
+
+      const prop1 = 'prop1';
+      const prop2 = 'prop2';
+
+      const createTransaction = (prop, turn = 0) => {
+        return firebase.firestore().runTransaction(async transaction => {
+          const doc = await transaction.get(docRef);
+          const data = doc.data();
+
+          if (data.turn !== turn) {
+            throw new Error('turn cannot exceed 1');
+          }
+
+          const update = {
+            turn: turn + 1,
+            [prop]: 1,
+          };
+
+          transaction.update(docRef, update);
+        });
+      };
+
+      const promises = [createTransaction(prop1), createTransaction(prop2)];
+
+      try {
+        await Promise.all(promises);
+      } catch (e) {
+        // do nothing
```

Comment:
Should explicitly check for aborted transaction errors, anything else is a test failure